packages/amplify_authenticator/lib/src/widgets/authenticator_banner.dart

```diff
@@ -21,46 +21,87 @@ import 'package:flutter/material.dart';
 MaterialBanner createMaterialBanner(
   BuildContext context, {
   required StatusType type,
-  required Widget content,
-  required List<Widget> actions,
+  required String message,
+  required Function() actionCallback,
 }) {
-  final margin = MediaQuery.of(context).viewPadding.top;
+  final theme = Theme.of(context);
+
+  // foreground and background colors are based on material standards for text legibility
+  // if they are not provided by the theme. See https://material.io/design/color/text-legibility.html#text-backgrounds
+
+  // fallback on default color based on type.
+  final backgroundColor = theme.bannerTheme.backgroundColor ?? type.color;
```

Comment:
My DRY sense are tingling when I see all this duplicated code. If you create a type to wrap backgroundColor/foregroundColor you could create a `ColorChoices Function(customForegroundColor?, customBackgroundColor?)` method to isolate this logic